GFS_layer/GFS_physics_driver.F90

```diff
@@ -617,6 +619,16 @@ subroutine GFS_physics_driver                         &
         dq3dt_initial = Diag%dq3dt
       endif
       
+      if (Model%override_surface_radiative_fluxes) then
+        adjsfcdlw_for_lsm => Statein%adjsfcdlw_override
+        adjsfcdsw_for_lsm => Statein%adjsfcdsw_override
+        adjsfcnsw_for_lsm => Statein%adjsfcnsw_override
```

Comment:
Where is Statein being populated with these override values.  While the ai2cm version of the model may have used Statein for this purpose, I'm not sure that's something we want to proliferate if SHiELD is to have a long life at GFDL.  I'd prefer to see a new type in GFS for data overrides or use the Coupler type which will eventually be used to move data back and forth to a decoupled land model.  I'll let @lharris4 make the call on this though.

Reply:
Thanks for taking a look @bensonr.  Indeed this could be made clearer; since the wrapper is being used to populate the override values it will always look a bit mysterious from the fortran perspective.  We basically just need some allocated state in the fortran model that the wrapper can point to, which could be in `Statein`, a new type, or the `Coupler` type.  A new dedicated type (with a comment describing the intent) is maybe best since the use-case is quite different than anything else that exists currently.  Happy to take guidance from you and @lharris4 on this.

Reply:
I think the coupler type may be the best choice, if it already has these fields defined within it. The coupler will probably be getting heavier use soon as we start working on SHiELD coupling.

Reply:
BTW is `data_override()` available within the SHiELD Physics? This may be a better route since it is a standard, supported FMS functionality, and we are already using this for the SSTs.

Reply:
@spencerkclark will correct me if I'm wrong, but I believe these fields are coming from the Python-based ML/DA system directly and not via datafiles read and or interpolated.  So in this case, the data_override functionality from FMS would not be applicable here.

Reply:
> I think the coupler type may be the best choice, if it already has these fields defined within it. The coupler will probably be getting heavier use soon as we start working on SHiELD coupling.

Thanks @lharris4 this seems OK so long as it will not be confusing for anyone reading the code.  I'll try and make things clear with a comment.

> BTW is `data_override()` available within the SHiELD Physics? This may be a better route since it is a standard, supported FMS functionality, and we are already using this for the SSTs.

Indeed @bensonr beat me to it.  The tricky thing here is that the data we would like to use for overriding may not necessarily be read in from a file (as I understand `data_override` would require).  It instead can be predicted by an ML model as the simulation is running.

Reply:
I understand better now @bensonr @spencerkclark. This is different from the usual `data_override()` use case. Indeed going through the datatypes will be a better way. This will probably also be useful for various DA routines like what Naomi is setting up in LM4.